corgidrp/measure_companions.py

```diff
@@ -0,0 +1,516 @@
+    This function does the following:
+      1) Measure host star counts in a direct image.
+      2) Measure reference PSF counts (from off-axis calibration).
+      3) Compute ratio for scaling the off-axis PSF to represent the star at companion's location.
+      4) (Optional) Forward-model off-axis star image through PSF subtraction if method= 'forward_model'.
+      5) Measure the companion count ratio & apparent magnitude.
+
+    Args:
+        direct_star_image (corgidrp.data.Image): A direct (unocculted) image of the host star (in e-). 
+        psf_sub_image (corgidrp.data.Image): The final (or intermediate) PSF-subtracted image with companions
+            in e-, with .data, .hdr, .ext_hdr.
+        reference_psf (corgidrp.data.Image): An off-axis calibration PSF of a star at some known 
+            separation where mask effects are negligible (it is near 6 lam/D).
+        ct_dataset (corgidrp.data.Dataset, optional): A dataset containing some coronagraphic observations.
+        ct_cal (corgidrp.data.CoreThroughputCalibration, optional): A Core Throughput calibration file containing 
+            PSFs at different distances from the mask and their corresponding positions and throughputs.
+        phot_method ({'aperture', 'gauss2d'}, optional): Photometry method for measuring 
+            companion counts on the (already) PSF-subtracted image.
+        coronagraphic_dataset (corgidrp.data.Dataset, optional): Coronagraphic images with companions. 
+        fluxcal_factor (corgidrp.Data.FluxcalFactor, optional): Fluxcal factor with .ext_hdr['ZP'].
+        FpamFsamCal (corgidrp.data.FpamFsamCal, optional): FPAM to EXCAM transformation matrix.
+        host_star_in_calspec (bool, optional):
+        forward_model (bool, optional):
+        output_dir (str, optional): Output directory path.
+        verbose (bool, optional): If True, print progress messages.
+
+    Returns:
+        result_table (astropy.table.Table):A table with the following columns (for checking now, replace later
+            with regular returns):
+                - id : companion label
+                - x, y : location in pixels
+                - counts_raw : companion counts in e- (post-subtraction)
+                - counts_err : estimated counts uncertainty
+                - counts_corr : counts after throughput correction
+                - counts_ratio : counts_corr / host_star_counts (if host_star_counts is given)
+                - mag : apparent magnitude (computed via host_star_apmag or zero point or fluxcal_factor)
+    """
+    phot_method_kwargs = get_photometry_kwargs(phot_method)
+
+    # i. Select reference PSF: assume off-axis PSF with highest CT has negligible effect from the mask 
+    # (it is near 6 lam/D see Figs. 9 and 11 in John Krist's paper), or use simulated PSF w/o FPM masks
+   
+    # ii. Measure host star counts from direct image 
+    host_star_counts, _ = measure_counts(direct_star_image, phot_method, None, **phot_method_kwargs)
```

Comment:
if the direct_star_image is taken with the ND (almost certainly), probably need to correct for the transmission of the ND first, before anything else.